src/main/java/com/cloudbees/jenkins/plugins/bitbucket/api/BitbucketBuildStatus.java

```diff
@@ -72,7 +73,7 @@ public BitbucketBuildStatus(String hash, String description, String state, Strin
         this.description = description;
         this.state = state;
         this.url = url;
-        this.key = key;
+        this.key = DigestUtils.md5Hex(key);
```

Comment:
why don't you check for the min length anymore? I think the hash should only be the fallback.